givelively/exception_notifier
# Allow passing of additional parameters to Raven.capture_exception

Jira ticket: [DEV-536](https://give-lively.atlassian.net/browse/DEV-536)


This is needed for givelively/charity-api#4594 (part of [DEV-364](https://give-lively.atlassian.net/browse/DEV-364))

`ExceptionNotifier.call` currently only accepts a single parameter if you pass it an Exception object. However, `Raven.capture_exception` supports passing extra parameters, such as `extra`.

This PR simply allows calling `ExceptionNotifier.call` the same way. For example, [this](https://github.com/givelively/charity-api/blob/1edbb527ad81370f48d618deb6b709ef2555dcf3/app/controllers/v2/paypal_charges_controller.rb#L14) is how we're going to be using it when handling failed PayPal API calls in the associated PR:

```
ExceptionNotifier.call(e, extra: { response: e.try(:response) })
```
## QA Notes

#### Steps to QA



1. Set `SENTRY_DSN` in your .env file to [the value seen here](https://sentry.io/settings/give-lively/projects/charity-api/keys/)
1. Remove `gem 'exception_notifier', '0.1.1', git: 'https://github.com/givelively/exception_notifier/'` from your Gemfile in charity-api
2. Replace it with `gem 'exception_notifier', path: '../exception_notifier'`
3. Run `bundle install`
4. Open up a Rails console
5. Run `ExceptionNotifier.call(Exception.new('Testing exception_notifier PR #5'), extra: { foo: 'bar' })`
6. View the exception in [Sentry](https://sentry.io/organizations/give-lively/issues/1698262126/events/e900cc4e0f5e444baa56dfca6a6b0ff5/?project=2530079)
7. Scroll down to "Additional Data"
8. Make sure `foo` is set to `bar`

When you're done testing this, remember to delete `SENTRY_DSN` from your .env file so we don't clutter Sentry with our local exceptions.






![image](https://user-images.githubusercontent.com/401283/83304213-75348d00-a1cc-11ea-9239-b185381c2d96.png)


#### Author checklist

- [x] Mention the Jira issue number at the very top of the PR description
- [x] Attach any relevant screenshots
- [x] Document any notes for QA ([Example QA notes](https://trello.com/c/PTe1CJ8C/3299-allow-all-npo-admins-to-set-a-customer-service-email-address-as-a-reply-to-for-outgoing-give-lively-email#comment-5c89417583773936e9f83a7d))
- [x] Assign someone to review the PR
- [ ] Move Jira issue to the `In Code Review` column

#### Reviewer checklist

- [ ] Review the PR
- [ ] Move the issue to the `Ready for QA` column once the PR is approved

## Helpful links

[Engineering issue & PR workflow/Lifecycle of an issue](https://github.com/givelively/charity-api/wiki/Lifecycle-of-an-issue)
